planet/api/order_details.py

```diff
@@ -13,722 +13,445 @@
 # the License.
 """Functionality for preparing order details for use in creating an order"""
 from __future__ import annotations  # https://stackoverflow.com/a/33533514
-import copy
-import json
 import logging
-from typing import List, Union
+from typing import List
 
 from .. import geojson, specs
 
 LOGGER = logging.getLogger(__name__)
 
 
-class OrderDetailsException(Exception):
-    """Exceptions thrown by OrderDetails"""
-    pass
-
-
-class OrderDetails():
-    '''Validating and preparing order details for submission.
-
-    Can be built up from order detail parts:
+def build_request(
+        name: str,
+        products: List[dict],
+        subscription_id: int = 0,
+        delivery: dict = None,
+        notifications: dict = None,
+        order_type: str = None,
+        tools: List[dict] = None
+) -> dict:
+    '''Prepare an order request.
 
     ```python
-    >>> from planet.api.order_details import OrderDetails, Product
-    >>>
-    >>> image_ids = ['3949357_1454705_2020-12-01_241c']
-    >>> order_detail = OrderDetails(
-    ...     'test_order',
-    ...     [Product(image_ids, 'analytic', 'psorthotile')]
-    ... )
+    >>> from planet.api.order_details import (
+    ...     build_request, product, toar_tool, reproject_tool, tile_tool)
+    ...
+    >>> products = [
+    ...     product(['20170614_113217_3163208_RapidEye-5'],
+    ...             'analytic', 'REOrthoTile')])
+    ... ]
+    ...
+    >>> tools = [
+    ...     toar_tool(scale_factor=10000),
+    ...     reproject_tool(projection='WSG84', kernel='cubic'),
+    ...     tile_tool(1232, origin_x=-180, origin_y=-90,
+    ...               pixel_size=0.000027056277056,
+    ...               name_template='C1232_30_30_{tilex:04d}_{tiley:04d}')
+    ... ]
+    ...
+    >>> order_request = build_request(
+    ...     'test_order', products, tools)
     ...
 
     ```
 
-    or from a dict describing the order detail:
+    Parameters:
+        name: Name of the order.
+        products: Product(s) from the Data API to order.
```

Comment:
this is an extremely minor comment, but would it be nicer to define `products` strictly in terms of item IDs & asset types? In which case we could avoid the possibly-confusing reference to Data API completely

Reply:
i agree fully! the question is whether we want to follow the naming convention of the orders api